sdk/azidentity/managed_identity_client.go

```diff
@@ -182,14 +185,18 @@ func (c *managedIdentityClient) createAppServiceAuthRequest(ctx context.Context,
 		request.Header.Set("secret", os.Getenv(msiSecret))
 		q.Add("api-version", "2017-09-01")
 		q.Add("resource", strings.Join(scopes, " "))
+		if clientID != "" {
+			q.Add("clientid", clientID)
```

Comment:
Might want to add a comment explaining why this is different than 20190801

Reply:
done